ObjCCBOR/CBOR.h

```diff
@@ -0,0 +1,44 @@
+//
+//  CBOR.h
+//  ObjCCBOR
+//
+//  Created by Hamilton Chapman on 20/02/2020.
+//  Copyright © 2020 Ditto. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface CBOR : NSObject
+
+/**
+ Decode CBOR data into its corresponding NSObject type. Note that
+ this library supports primitives so the resulting type could be any
+ of `NSDictionary`, `NSArray`, `NSString`, `NSNumber`, `NSNull` or
+ `NSData`.
+
+ @param data The data to be decoded.
+ @param error An optional error out parameter. If not `NULL`, then
+ an error object will be written to this parameter should decoding fail.
+ @returns Returns one of `NSDictionary`, `NSArray`, `NSString`,
+ `NSNumber`, `NSNull` or `NSData`. Will return `nil` if the decoding
+ fails.
+ */
++ (nullable id)decode:(NSData *)data error:(NSError * _Nullable __autoreleasing *)error;
```

Comment:
```suggestion
+ (nullable id)decodeData:(NSData *)data error:(NSError * _Nullable __autoreleasing *)error;
```

> Make the word before the argument describe the argument.
> 
> |Example||
> |-- | -- |
> | - (id)viewWithTag:(NSInteger)aTag; | Right. |
> | - (id)taggedView:(int)aTag; | Wrong. |

[Coding Guidelines for Cocoa](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html)

---

When I looked a little closer, I realized `NSJSONSerialization` does this too.